src/core/job.c

```diff
@@ -1338,6 +1338,8 @@ void job_shutdown_magic(Job *j) {
         if (detect_container() > 0)
                 return;
 
+        manager_invalidate_startup_units(j->unit->manager);
```

Comment:
i see no reason to conditionalize this on detect_container(). if we have the perms to muck with cgroups, then i think we should do this in a container too if configured that way.

i.e. just move this line before the detect_container() check

Reply:
Yeah, absolutely. I missed this...